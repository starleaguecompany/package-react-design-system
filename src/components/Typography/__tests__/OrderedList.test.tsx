import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { OrderedList, ListItem } from '../index';

describe('OrderedList', () => {
  test('should render correctly with no props', () => {
    const { container } = render(<OrderedList />);

    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('ordered', { exact: true });
  });

  test('should render correctly with ListItem', () => {
    render(
      <OrderedList>
        <ListItem>Testing list</ListItem>
      </OrderedList>
    );

    expect(screen.getByRole('list')).toMatchSnapshot();
    expect(screen.getByRole('list')).toContainElement(screen.getByRole('listitem'));
    expect(screen.getByText('Testing list')).toBeVisible();
  });

  test('should render correctly with attributes', () => {
    render(
      <OrderedList id="test-id" className="test-class">
        Testing list
      </OrderedList>
    );

    expect(screen.getByRole('list')).toMatchSnapshot();
    expect(screen.getByRole('list')).toHaveAttribute('id', 'test-id');
    expect(screen.getByRole('list')).toHaveClass('test-class');
  });
});
