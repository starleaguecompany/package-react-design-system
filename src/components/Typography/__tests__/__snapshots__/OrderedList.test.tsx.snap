// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderedList should render correctly with ListItem 1`] = `
<ol
  class="ordered"
>
  <li
    class="item"
  >
    Testing list
  </li>
</ol>
`;

exports[`OrderedList should render correctly with attributes 1`] = `
<ol
  class="test-class ordered"
  id="test-id"
>
  Testing list
</ol>
`;

exports[`OrderedList should render correctly with no props 1`] = `
<div>
  <ol
    class="ordered"
  />
</div>
`;
